src/transformers/models/paligemma/modeling_paligemma.py

```diff
@@ -378,7 +378,7 @@ def _update_causal_mask(
             if is_training:
                 causal_mask = torch.triu(causal_mask, diagonal=1)
             else:
-                causal_mask = torch.zeros_like(causal_mask)
+                causal_mask[:, :sequence_length] = 0.0
```

Comment:
this was the cause as it was not masking dummy tokens from static cache, and thus we always ended up with no mask on those token positions